### PR TITLE
Meanshift clustering

### PR DESCRIPTION
**Pull request by JordanCheney**
This is an implementation of the meanshift clustering algorithm. It has the nice feature of not requiring the number of clusters to be known before clustering occurs.

@caotto could you review?

### CODE REVIEW

File: openbr/plugins/cluster/meanshift.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        QList<int> groups;
+        int newGroupIdx = 0;
+        foreach (const Mat &point, points) {
+            int group = nearestGroup(point, points, groups);
+            if (group < 0)
+                group = newGroupIdx++;
+            groups.append(group);
+        }
+
+        qDebug("created %d clusters from %d templates", newGroupIdx, points.size());
```
Comment (caotto) [inline openbr/plugins/cluster/meanshift.cpp]: please remove

File: openbr/plugins/cluster/meanshift.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        Mat distances(1, original_points.size(), CV_32FC1);
+        for (int i = 0; i < original_points.size(); i++)
+            distances.at<float>(0, i) = distance->compare(point, original_points[i]);
+
+        Mat weights = gaussian_kernel(distances, kernelBandwidth);
+        point = (weights * OpenCVUtils::toMat(original_points)) /  sum(weights)[0];
+    }
+
+    inline Mat gaussian_kernel(const Mat &distance, const float bandwidth)
```
Comment (caotto) [inline openbr/plugins/cluster/meanshift.cpp]: Do we have an official capitalization style policy, most of the rest of this is like gaussianKernel

**Comment by caotto** [comment]
Generally looks OK, please add documentation.

**Comment by JordanCheney** [comment]
Acceptable changes?

**Comment by caotto** [comment]
looks ok to me

**Pull request merged**